tests/flytekit/unit/interaction/test_click_types.py

```diff
@@ -498,3 +499,30 @@ class Datum:
     assert v.z["key"].b == "nested"
     assert v.w[0].a == 30
     assert v.w[0].b == "list_item"
+
+
+def test_pickle_type():
+    t = PickleParamType()
+    value = {"a": "b"}
+    v = t.convert(value=value, param=None, ctx=None)
+    assert v == value
+
+    with pytest.raises(click.BadParameter):
+        t.convert("", None, None)
+
+    with pytest.raises(click.BadParameter):
+        t.convert("module.x", None, None)
+
+    with pytest.raises(click.BadParameter):
+        t.convert("module:var", None, None)
+
+    with pytest.raises(click.BadParameter):
+        t.convert("typing:not_exists", None, None)
+
+    # test that it can load a variable from a module
+    with tempfile.NamedTemporaryFile("w", dir=".", suffix=".py", delete=True) as f:
+        f.write("a = 1")
+        f.flush()
+        # find the base name of the file
+        basename = os.path.basename(f.name).split(".")[0]
+        assert t.convert(f"{basename}:a", None, None) == 1
```

Comment:
Windows doesn't let the same file be opened twice, which causes this [permission denied issue](https://github.com/flyteorg/flytekit/actions/runs/11372614166/job/31637345293?pr=2815#step:7:2869). The easiest thing to do is to sequence the calls to open+close+unlink. Something like:
```suggestion
    tmp_f = tempfile.NamedTemporaryFile(dir=".", suffix=".py", delete=False)
    f = open(tmp_f.name, "w")
    f.write("a = 1")
    f.close()
    # find the base name of the file
    basename = os.path.basename(tmp_f.name).split(".")[0]
    assert t.convert(f"{basename}:a", None, None) == 1
    # Not sure if need to delete this file, it's a temp file anyway, but leaving here for completeness
    os.unlink(tmp_f.name)
```